src/qutip_jax/ode.py

```diff
@@ -14,12 +14,16 @@
 
 @jax.jit
 def _cplx2float(arr):
-    return jnp.stack([arr.real, arr.imag])
+    if jnp.iscomplexobj(arr):
+        return jnp.stack([arr.real, arr.imag])
+    return arr
 
 
 @jax.jit
 def _float2cplx(arr):
-    return arr[0] + 1j * arr[1]
+    if arr.ndim == 3:
+        return arr[0] + 1j * arr[1]
+    return arr
```

Comment:
Test

Reply:
It is covered in `test_non_cplx128_JaxQobjEvo`.